webapp/graphite/tags/migrations/0001_initial.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.11 on 2017-07-07 21:07
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Series',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('path', models.TextField(unique=True)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='SeriesTag',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('series', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='tags.Series')),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Tag',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('tag', models.TextField(unique=True)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='TagValue',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('value', models.TextField(unique=True)),
+            ],
+        ),
+        migrations.AddField(
+            model_name='seriestag',
+            name='tag',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='tags.Tag'),
```

Comment:
if multiple series have `tagX`, then one of them gets deleted, wouldn't this cascade to delete that `tagX` despite the fact that other series refer to it as well?